### PR TITLE
[github-actions] fix version workflow's name

### PR DESCRIPTION
**Pull request by simonlingoogle**

### CODE REVIEW

File: .github/workflows/version.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
-name: Build
+name: Version
```
Comment (jwhui) [inline .github/workflows/version.yml:29]: Can we be more descriptive? Something like "API Version"?

Comment (simonlingoogle) [reply .github/workflows/version.yml:29]: The yml file was intentionally named to be `version.yml`. 
@bukepo described about it in https://github.com/openthread/openthread/pull/5034#discussion_r433592601
So I think the name should also be `Version` for the same reason.

Comment (jwhui) [reply .github/workflows/version.yml:29]: We have a lot of versions, some are not related to API, like Thread 1.1/1.2. I think we should be clear about what this workflow actually checks. @bukepo @simonlingoogle , thoughts?

Comment (bukepo) [reply .github/workflows/version.yml:29]: I was considering further extensions, for example when we finally have some lts versions, say 1.0, we may add checks for it interop with master. And we may add it in this workflow. However, I am fine with renaming it to API version given there is only a single check for now.

Comment (simonlingoogle) [inline .github/workflows/version.yml:29]:
```suggestion
name: API Version
```
Let's use `API Version` for now.

Comment (simonlingoogle) [reply .github/workflows/version.yml]: renamed

**Pull request merged**